### PR TITLE
API-38762-add-check-for-error

### PR DESCRIPTION
**Pull request by rockwellwindsor-va**
## Summary
* Updates to set_evss_response method
	* Adds a check to see if the error responds to `.original_body` before calling it
	* Since the logic moves in one layer there is also a default catch added as a safety
* Refactors based on `pronto/reek` 
      * Moves the respond_to? check for original_body into a method
      * refactors the `set_evss_response` to eliminate repeated code
      * changes variable name from `e` to `err`

## Related issue(s)
[API-38762](https://jira.devops.va.gov/browse/API-38762)

## Testing done
- [x] *New code is covered by unit tests*

#### Testing Notes
* Currently, locally, the error can be generated using test user with ICN `1012666182V203559 ` and submitting a claim (via the async endpoint)
* BECAUSE IT IS ASYNC - Change the PDF generator call in the async workflow to `new.perform` and do the same wit the docket container job call in the pdf generator sidekiq job.
    *  Without the check on `service_base.rb` line 65 it will error out when the error response get to that point.
    *  To test the error message format you can hit the same endpoint, add a byebug in the pdf generator and get the claim ID and then check the v2 claims show for that claim to verify the source is present.  (You can see it return null by reverting the change in the `Lighthouse::BackendServiceException` file).
#### Adding check
* Given the `form526.submit.establishClaim.serviceError` when that arrived at the `set_evss_response` method in `service_base.rb` it was erroring on the line `error&.original_body.present?` because `error` is present but does not have an `original_body` so this just adds a `responds_to` check before calling the method.

## What areas of the site does it impact?
	modified:   modules/claims_api/app/sidekiq/claims_api/service_base.rb
	modified:   modules/claims_api/lib/claims_api/common/exceptions/lighthouse/backend_service_exception.rb
	modified:   modules/claims_api/spec/sidekiq/service_base_spec.rb

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [x]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: modules/claims_api/app/sidekiq/claims_api/service_base.rb [hunk 1]
```diff
@@ -61,13 +61,24 @@ def save_auto_claim!(auto_claim, status)
...
+          # original_body method but still not have it
+          errors_to_add << error
+        end
       elsif error&.errors.present?
-        error&.errors&.each { |e| auto_claim.evss_response << e }
+        errors_to_add.concat(error.errors)
       end
 
+      # Add all collected errors to the auto_claim evss_response
+      auto_claim.evss_response.concat(errors_to_add)
```
[the remark this answers is not in the document]
Comment (rockwellwindsor-va) [reply modules/claims_api/app/sidekiq/claims_api/service_base.rb:80]: I cannot see a way around this, obviously totally open to suggestions

File: modules/claims_api/lib/claims_api/common/exceptions/lighthouse/backend_service_exception.rb [hunk 2]
```diff
@@ -14,6 +14,7 @@ def errors
             errors_array = []
             @errors.each do |err|
               errors_array << {
+                key: err[:key] || nil,
```
[the remark this answers is not in the document]
Comment (rockwellwindsor-va) [reply modules/claims_api/lib/claims_api/common/exceptions/lighthouse/backend_service_exception.rb:17]: This does point to a larger refactoring of these so they are not responsible for formatting the errors too, but that is outside the scope of this PR I suspect.  Again totally open to suggestions on that,  I think doing that would blow this out a little bit.

File: modules/claims_api/app/sidekiq/claims_api/service_base.rb [hunk 3]
```diff
@@ -61,18 +61,29 @@
 
     def set_evss_response(auto_claim, error)
       auto_claim.evss_response ||= []
-
-      if error&.original_body.present?
-        error&.original_body&.each { |e| auto_claim.evss_response << e }
+      errors_to_add = []
+
+      if error_responds_to_original_body?(error)
```
[the remark this answers is not in the document]
Comment (rockwellwindsor-va) [reply modules/claims_api/app/sidekiq/claims_api/service_base.rb:66]: Cannot see a way around this, totally open to suggestions

File: modules/claims_api/app/sidekiq/claims_api/service_base.rb [hunk 4]
```diff
@@ -61,18 +61,29 @@
...
+        if error&.original_body.present?
+          errors_to_add.concat(error.original_body)
```
[the remark this answers is not in the document]
Comment (rockwellwindsor-va) [reply modules/claims_api/app/sidekiq/claims_api/service_base.rb:68]: Cannot see a way to address this but open to feedback on that

File: modules/claims_api/app/sidekiq/claims_api/service_base.rb [hunk 5]
```diff
@@ -61,18 +61,29 @@ def save_auto_claim!(auto_claim, status)
...
+          # original_body method but still not have it
+          errors_to_add << error
+        end
       elsif error&.errors.present?
-        error&.errors&.each { |e| auto_claim.evss_response << e }
+        errors_to_add.concat(error.errors)
       end
 
+      # Add all collected errors to the auto_claim evss_response
+      auto_claim.evss_response.concat(errors_to_add)
```
[the remark this answers is not in the document]
Comment (rockwellwindsor-va) [reply modules/claims_api/app/sidekiq/claims_api/service_base.rb:80]: Not sure exactly what that kind of refactor would entail.

**Review by mchristiansonVA — APPROVED** [review]
Tested locally and confirmed expected results using test ICN indicated. Looks good to me!

**Pull request merged**